packages/util-retry/src/defaultRetryToken.ts

```diff
@@ -1,91 +1,47 @@
-import { RetryErrorInfo, RetryErrorType, StandardRetryBackoffStrategy, StandardRetryToken } from "@aws-sdk/types";
+import { RetryErrorInfo, RetryErrorType, StandardRetryToken } from "@aws-sdk/types";
 
-import {
-  DEFAULT_RETRY_DELAY_BASE,
-  MAXIMUM_RETRY_DELAY,
-  NO_RETRY_INCREMENT,
-  RETRY_COST,
-  THROTTLING_RETRY_DELAY_BASE,
-  TIMEOUT_RETRY_COST,
-} from "./constants";
-import { getDefaultRetryBackoffStrategy } from "./defaultRetryBackoffStrategy";
-
-/**
- * @public
- */
-export interface DefaultRetryTokenOptions {
-  /**
-   * The total amount of retry tokens to be decremented from retry token balance.
-   */
-  retryCost?: number;
-
-  /**
-   * The total amount of retry tokens to be decremented from retry token balance
-   * when a throttling error is encountered.
-   */
-  timeoutRetryCost?: number;
-
-  /**
-   *
-   */
-  retryBackoffStrategy?: StandardRetryBackoffStrategy;
-}
+import { MAXIMUM_RETRY_DELAY, RETRY_COST, TIMEOUT_RETRY_COST } from "./constants";
 
 /**
  * @internal
  */
-export const getDefaultRetryToken = (
-  initialRetryTokens: number,
-  initialRetryDelay: number,
-  initialRetryCount?: number,
-  options?: DefaultRetryTokenOptions
-): StandardRetryToken => {
-  const MAX_CAPACITY = initialRetryTokens;
-  const retryCost = options?.retryCost ?? RETRY_COST;
-  const timeoutRetryCost = options?.timeoutRetryCost ?? TIMEOUT_RETRY_COST;
-  const retryBackoffStrategy = options?.retryBackoffStrategy ?? getDefaultRetryBackoffStrategy();
-
-  let availableCapacity = initialRetryTokens;
-  let retryDelay = Math.min(MAXIMUM_RETRY_DELAY, initialRetryDelay);
-  let lastRetryCost: number | undefined = undefined;
-  let retryCount = initialRetryCount ?? 0;
-
-  const getCapacityAmount = (errorType: RetryErrorType) => (errorType === "TRANSIENT" ? timeoutRetryCost : retryCost);
+export const createDefaultRetryToken = ({
```

Comment:
Throw errors in implementations of deprecated functions instead?

That way you don't need to pass `availableCapacity` for creating default RetryToken.

Reply:
I think we should remove the functions altogether from the interface.

It's very unlikely that someone would have written subclass of StandardRetryStrategy which doesn't used methods from parent class. And if they do use methods from parent class, they'll unlikely be calling functions on StandardRetryToken which modifies availableCapacity.

Reply:
We're already removing retryToken member from `StandardRetryStrategy` class, as it should not exist.
The only way to access retryToken is by calling `acquireInitialRetryToken` which is done by callee of the strategy, i.e. middleware.